lib/services/graph-progress.js

```diff
@@ -0,0 +1,230 @@
+// Copyright © 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+
+'use strict';
+
+module.exports = graphProgressServiceFactory;
+
+graphProgressServiceFactory.$provide = 'Services.GraphProgress';
+graphProgressServiceFactory.$inject = [
+    '_',
+    'Protocol.Events',
+    'Services.Waterline',
+    'Logger',
+    'Constants',
+    'Promise',
+    'Assert',
+    'GraphProgress'
+];
+
+function graphProgressServiceFactory(
+    _,
+    eventsProtocol,
+    waterline,
+    Logger,
+    Constants,
+    Promise,
+    assert,
+    GraphProgress
+) {
+    var logger = Logger.initialize(graphProgressServiceFactory);
+
+    function GraphProgressService() {
+    }
+
+    /**
+     * Publish a graph started progress event over AMQP.
+     *
+     * @param {Object} graph
+     * @returns {Promise}
+     * @memberOf GraphProgressService
+     */
+    GraphProgressService.prototype.publishGraphStarted = function(graph) {
+        assert.object(graph, 'graph');
+        assert.string(graph.name , 'graph.name');
+
+        var progress = GraphProgress.create(graph, 'Graph "' + graph.name + '" started');
+        return eventsProtocol.publishProgressEvent(
+            graph.instanceId,
+            progress.getProgressEventData()
+        )
+        .catch(function(error) {
```

Comment:
I'd suggest don't let the swallow error as the default behavior, it's better to throw error, with some flag to enable swallow errors, example below:

```javascript
GraphProgressService.prototype.publishGraphStarted = function(graph, options) {
  ...
  return eventsProtocol.publishProgressEvent(....)
  .catch(function(err) {
    if (options.swallowError) {
        log.error(...);
    } else {
      throw err;
    }
  });
}
```

So if user wants to swallow error, it called by:
```javascript
publishGraphStarted(graph, { swallowError: true });
```